web/packages/teleterm/src/ui/services/deepLinks/deepLinksService.ts

```diff
@@ -199,6 +199,9 @@ export class DeepLinksService {
       }
     }
 
+    // fetch the rootCluster now that we are logged in
+    rootCluster = this.clustersService.findCluster(rootClusterUri);
```

Comment:
Hmm, fetching the root cluster at this point doesn't guarantee that the user has logged in. If the given root cluster already exists in the state when we enter `loginAndSetActiveWorkspace`, but the cert has expired, you'll get a `Cluster` object with `connected: false`.

This makes me think we should not return `rootCluster: Cluster` from this function at all. Instead, we could return merely `rootClusterUri`. If any particular deep link handler needs a `Cluster` object, it can fetch one after inspecting `isAtDesiredWorkspace` – this way we'd be sure that at this point the user has logged in to the cluster and we have full cluster details.

Reply:
Thats fair. updated https://github.com/gravitational/teleport/pull/40802/commits/bcb13e21ac769ed7cdbd3019a2e4272d83654e6c